internal/clientconn/conninfo/conn_info.go

```diff
@@ -34,6 +36,16 @@ type ConnInfo struct {
 	rw       sync.RWMutex
 	username string
 	password string
+
+	curRW  sync.RWMutex
```

Comment:
Asking for educational purposes 🤓  I see `RWMutex` is set, could you explain the logic about reading/writing? I see in the code below, it uses write locks to set/remove the cursor, but I don't understand how `sync.RWMutex` benefits over `sync.Mutex`.

Reply:
As far as I can understand there could be `n` readers with the `RWMutex` apart from the regular `Mutex`.

Reply:
Why can't a regular mutex be used with those `n` readers?